### ISSUE TITLE
master: datatype/opal_datatype_pack failures in ibm/pt2pt/sendrecv_big

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#6221]
MTT is showing a bunch of failures on master with datatype packing errors -- I think with large messages.  For example, ibm/pt2pt/sendrecv_big (https://mtt.open-mpi.org/index.php?do_redir=3158 -- see below).

I tried to git bisect this, but the same failure was happening on master back of December of 2016!  (I went back as far as 5fb3e0a9b696622511e8c557aa77061991e188a8 before giving up)

@bosilca Can you have a look?  Is the test itself wrong?

```
[mpi014:31043] ../../opal/datatype/opal_datatype_pack.h:53
	Pointer 0x2ac3c9713010 size 16352 is outside [0x2aaac0c03010,0x2ab90edc3010] for
	base ptr 0x2aaac0c03010 count 1 and data 
[mpi014:31043] Datatype 0x70e7c0[] size 30720000000 align 4 id 0 length 7 used 6
true_lb 0 true_ub 61440000000 (true_extent 61440000000) lb 0 ub 61440000000 (extent 61440000000)
nbElems -909934592 loops 4 flags 104 (committed )-c-----GD--[---][---]
   contain OPAL_FLOAT4:* 
--C--------[---][---]    OPAL_LOOP_S 192 times the next 2 elements extent 80000000
--C---P-D--[---][---]    OPAL_FLOAT4 count 20000000 disp 0xaba950000 (46080000000) blen 0 extent 4 (size 80000000)
--C--------[---][---]    OPAL_LOOP_E prev 2 elements first elem displacement 46080000000 size of data 80000000
--C--------[---][---]    OPAL_LOOP_S 192 times the next 2 elements extent 80000000
--C---P-D--[---][---]    OPAL_FLOAT4 count 20000000 disp 0x0 (0) blen 0 extent 4 (size 80000000)
--C--------[---][---]    OPAL_LOOP_E prev 2 elements first elem displacement 0 size of data 80000000
-------G---[---][---]    OPAL_LOOP_E prev 6 elements first elem displacement 46080000000 size of data 655228928
Optimized description 
-cC---P-DB-[---][---]     OPAL_UINT1 count -1819869184 disp 0xaba950000 (46080000000) blen 1 extent 1 (size 15360000000)
-cC---P-DB-[---][---]     OPAL_UINT1 count -1819869184 disp 0x0 (0) blen 1 extent 1 (size 15360000000)
-------G---[---][---]    OPAL_LOOP_E prev 2 elements first elem displacement 46080000000
[mpi014:31043] *** Process received signal ***
[mpi014:31043] Signal: Segmentation fault (11)
[mpi014:31043] Signal code: Address not mapped (1)
[mpi014:31043] Failing at address: 0x2ac3c9713010
[mpi014:31043] [ 0] /lib64/libpthread.so.0[0x35f9e0f710]
[mpi014:31043] [ 1] /lib64/libc.so.6(memcpy+0x15b)[0x35f9a89aab]
[mpi014:31043] [ 2] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libopen-pal.so.0(+0x8e6fc)[0x2aaaab4cc6fc]
[mpi014:31043] [ 3] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libopen-pal.so.0(opal_generic_simple_pack+0x1f7)[0x2aaaab4cdfc0]
[mpi014:31043] [ 4] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libopen-pal.so.0(opal_convertor_pack+0x323)[0x2aaaab4bfd79]
[mpi014:31043] [ 5] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libopen-pal.so.0(+0x11ed6a)[0x2aaaab55cd6a]
[mpi014:31043] [ 6] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x26fd22)[0x2aaaaad1cd22]
[mpi014:31043] [ 7] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(mca_pml_ob1_send_request_schedule_once+0x297)[0x2aaaaad1f7c5]
[mpi014:31043] [ 8] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x266767)[0x2aaaaad13767]
[mpi014:31043] [ 9] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x2667c8)[0x2aaaaad137c8]
[mpi014:31043] [10] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(mca_pml_ob1_recv_frag_callback_ack+0x21c)[0x2aaaaad14c6f]
[mpi014:31043] [11] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libopen-pal.so.0(+0x11ee55)[0x2aaaab55ce55]
[mpi014:31043] [12] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x26a717)[0x2aaaaad17717]
[mpi014:31043] [13] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(mca_pml_ob1_recv_request_ack_send_btl+0xd1)[0x2aaaaad18da3]
[mpi014:31043] [14] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x26b435)[0x2aaaaad18435]
[mpi014:31043] [15] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x26c02c)[0x2aaaaad1902c]
[mpi014:31043] [16] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(mca_pml_ob1_recv_request_progress_rndv+0xd5)[0x2aaaaad19eb1]
[mpi014:31043] [17] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x2689f6)[0x2aaaaad159f6]
[mpi014:31043] [18] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x2688d7)[0x2aaaaad158d7]
[mpi014:31043] [19] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(mca_pml_ob1_recv_frag_callback_rndv+0x69)[0x2aaaaad149de]
[mpi014:31043] [20] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libopen-pal.so.0(+0x11ee55)[0x2aaaab55ce55]
[mpi014:31043] [21] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x26fb64)[0x2aaaaad1cb64]
[mpi014:31043] [22] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(mca_pml_ob1_send_request_start_rndv+0x16f)[0x2aaaaad1f1b6]
[mpi014:31043] [23] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x260f51)[0x2aaaaad0df51]
[mpi014:31043] [24] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(+0x26109e)[0x2aaaaad0e09e]
[mpi014:31043] [25] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(mca_pml_ob1_send+0x5ad)[0x2aaaaad0f237]
[mpi014:31043] [26] /home/mpiteam/scratches/community/2018-12-21cron/KQq8/installs/pqJ3/install/lib/libmpi.so.0(PMPI_Sendrecv+0x46f)[0x2aaaaab9f16f]
[mpi014:31043] [27] pt2pt/sendrecv_big[0x400cdb]
[mpi014:31043] [28] /lib64/libc.so.6(__libc_start_main+0xfd)[0x35f9a1ed1d]
[mpi014:31043] [29] pt2pt/sendrecv_big[0x400a19]
[mpi014:31043] *** End of error message
***--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
-------------------------------------------------------------------------- Warning ::
pmix_list_remove_item - the item 0x2aaaac037c20 is not on the list 0x2aaaab456608 
mpirun: server/pmix_server.c:2342: _mdxcbfunc: Assertion `((0xdeafbeedULL << 32) + 0xdeafbeedULL) == ((pmix_object_t *) (tracker))->obj_magic_id' failed.
```

There's PMIx failures at the end of a bunch of these, but that appears to be a separate issue.  I'll follow up separately about them.

**Comment by jsquyres** [comment]
Remember to re-enable Cisco's MTT of this test when this is fixed (had to disable it because each fail was dropping a 9+GB corefile, and the sheer number of them was filling up my filesystem).

**Comment by ggouaillardet** [comment]
@jsquyres this issue is fixed in #6029 that has yet to be merged.

This test was introduced recently in order to make sure the fix has been merged and backported to all the release branches.

From a MTT point of view, could you instead set the max core size on a per test basis ?

**Comment by jsquyres** [comment]
We don't currently have an MTT control to limit the corefile size on a per-test basis.
